File: server/app/model/database/question.ts
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { Document } from 'mongoose';
import { Choice, choiceSchema } from './choice';
import { EstimatedParameters, estimatedParametersSchema } from './estimated-parameters';

export type QuestionDocument = Question & Document;

@Schema()
export class Question {
    @Prop({ required: true })
    id: string;

    @Prop({ required: true })
    type: string;

    @Prop({ required: true })
    text: string;

    @Prop({ required: true })
    points: number;

    @Prop({ type: [choiceSchema], default: [], required: true })
    choices: Choice[];

    @Prop({ type: [estimatedParametersSchema], default: {}, required: true })
    estimatedParameters: EstimatedParameters;

    @Prop({ required: false })
    lastModification: Date;
}

export const questionSchema = SchemaFactory.createForClass(Question);

File: client-mobile/app/src/main/java/com/example/polyquiz/match/domain/Question.kt
package com.example.polyquiz.match.domain

import java.util.Date

data class Question(
    val id: String,
    val type: String,
    var text: String,
    val points: Int,
    val choices: List<Choice>? = null,
    val estimatedParameters: EstimatedParameters? = null,
    val answer: String? = null,
    val lastModification: String?,
    val pictureUrl: String,
    val creatorName: String,
)
